parcels/particle.py

```diff
@@ -123,6 +124,10 @@ def __init__(self):
                 lat = self.getInitialValue(ptype, name='lat')
                 depth = self.getInitialValue(ptype, name='depth')
                 time = self.getInitialValue(ptype, name='time')
+                if time is None:
```

Comment:
I don't fully understand yet, where do we put a logger.error and where do we raise an error. Is it some rule for the moment?

Reply:
Ideally we want to move to the `parcels.logger` module everywhere. But we still have some legacy `raise` statements, from before the `parcels.logger` got introduced. Also, `parcels.logger.error()` does not yet throw an `exit` by itself, as one would expect. Would be good to clean this up at some point (not this PR though)?

Reply:
Such cleaning is not related to this PR. It should be done in another one I think